hoomd/hpmc/update.py

```diff
@@ -466,72 +467,51 @@ def N(self):  # noqa: N802 - allow N as a function name
         return N_dict
 
 
-class remove_drift:  # noqa - will be rewritten for v3
+class RemoveDrift(Updater):
     """Remove the center of mass drift from a system restrained on a lattice.
 
     Args:
-        mc (:py:mod:`hoomd.hpmc.integrate`): MC integrator.
-        external_lattice (:py:class:`hoomd.hpmc.field.lattice_field`): lattice
-          field where the lattice is defined.
-        period (int): the period to call the updater
-
-    The command hpmc.update.remove_drift sets up an updater that removes the
-    center of mass drift of a system every period timesteps,
+        reference_positions ((*N*, 3) `numpy.ndarray` of ``numpy.float32``): the
```

Comment:
The API should not depend on the build configuration. Don't use `Scalar` for datatypes exposed to the API, use `double`.

Reply:
@joaander Do you mean that the `reference_positions` should also be stored internally as `double` instead of `Scalar`?  My confusion comes from the observation that there's other cases of exported properties to the python API that are stored internally as `Scalar`. Or perhaps  I'm misunderstanding?

Reply:
Does the param dict reference the array by copy or by reference?

Reply:
I'm not entirely sure. @b-butler is better suited to answer this.

Reply:
OK, I looked at the code. `getReferencePositions` returns a copy. So you can modify `setReferencePositions` and `getReferencePositions` to work with `double` explicitly even while the internal data array is stored as a `Scalar` (floats can be cast to double with no data loss). This will keep the API consistent going forward when we enable mixed precision computations and the internal buffer will also be in double - but you don't need to make that change now.

Reply:
Ah, I see what you mean now. Thanks

Reply:
Then you'll also need to update the documentation to `float64` as the data type.